aten/src/ATen/core/ivalue.h

```diff
@@ -717,6 +721,12 @@ inline optional<T> IValue::toOptional() {
   return this->to<T>();
 }
 
+// Check whether the values are the same
+// This function is stricter than isSameIdentity, since this function
+// will return false if the tags don't match, however, this is not true
+// in isSameIdentity when some operand is None type
+bool operator==(const IValue& lv, const IValue& rv);
```

Comment:
I wonder whether we should have a method (something like deep_equal) instead. It'd be easy to misuse because it's unclear that comparison actually does deep comparison. @zdevito - what do you think?